lib/sqlalchemy/sql/selectable.py

```diff
@@ -2075,13 +2075,14 @@ def _init(
         recursive=False,
         nesting=False,
         _cte_alias=None,
-        _restates=(),
+        _restates=None,
```

Comment:
ok, above in the cache spec:

    ("_restates", InternalTraversal.dp_clauseelement_list),

has to change to

    ("_restates", InternalTraversal.dp_clauseelement),

Reply:
Should I declare `nesting` there as well ?

Reply:
probably.  it's not very easy for it to make a difference but yes, it should be declared with a dp_boolean